Automattic/jetpack
# Conversation: implement show/hide dialogue blocks system

_This PR is part of the [follow-up tasks](https://github.com/Automattic/jetpack/issues/18123) after merging `conversation`/`dialogue` blocks._

This PR implements a system to show and hide the `dialogue` blocks depending on the context of the current block. When the current block is a child of  `conversation` block, or child of children, or so on, the `dialogue` blocks shows. Otherwise, it hides the blocks.
It's an alternative approach to the template lock systems, where its lack to lock blocks in special conditions leads us to attempt this system.

Fixes #

#### Changes proposed in this Pull Request:

* Conversation: implement show/hide dialogue blocks system

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



No

#### Testing instructions:






* Start to edit a new blank post
* Try to add a `dialogue` block. It shouldn't appear as an insertion option 
 
<img src="https://user-images.githubusercontent.com/77539/102816811-63efdf80-43ad-11eb-9f84-6961b93b293d.png" width="300px" />

* Create a conversation block. You should see the default template.
<img src="https://user-images.githubusercontent.com/77539/102816927-93065100-43ad-11eb-8354-1fa071a7a9bf.png" width="500px" />

* and you should be able to add dialogue blocks there
<img src="https://user-images.githubusercontent.com/77539/102816984-aa453e80-43ad-11eb-9321-264aaa0d63f7.png" width="500px" />

* and also, you should be able to create content composing. The following is an example using Media&Text, which contains a `dialogue` block.

canvas | Outline
------|-------
![image](https://user-images.githubusercontent.com/77539/102817858-4b80c480-43af-11eb-9097-1dfc8b27af2d.png) | ![image](https://user-images.githubusercontent.com/77539/102817948-70753780-43af-11eb-92d4-05c1a1877559.png)



* 


#### Proposed changelog entry for your changes:


* Conversation: implement show/hide dialogue blocks system
